### PR TITLE
Jetpack connect: Remove redux stateful redirections

### PR DESCRIPTION
**Pull request by sirreal**
This PR removes redux-stateful redirections `goBackToWpAdmin` and `goToXmlrpcErrorFallbackUrl` from Jetpack Connect.

These redux-stateful redirections only served to set a flag indicating redirection is in flight. It was up to consumers to respect the flag.

The redirection logic has been moved to the consuming components.

✅  ~Includes #21103 which isolates an alphabteization commit. For the "relevant bits", see 09553f8...9b457df~

## Follow-up

Move retryAuth out of redux, state via query

## Testing
1. Verify logged-in auth flows continue to work as expected.
1. Verify xml-rpc fallback works as expected:
   1.  Set console debug for Calypso:
      ```js
      localStorage.debug = 'calypso:jetpack-connect:authorize-form'
      ```
   1. Get to the authorize step with the full auth query in the url
   1. Break the site's `xmlrpc.php` (`mv xmlrpc.php xmlrpc.php_`)
   1. Click authorize.
   1. Should see xmlrpc debug output on the console (`xmlrpc fallback to …`)
   1. Should still work

### CODE REVIEW

**Comment by sirreal** [comment]
Rebased to pull in #21103

File: client/jetpack-connect/authorize.js [hunk 1]
```diff
@@ -165,8 +157,15 @@ export class JetpackAuthorize extends Component {
 		} );
 	}
 
+	externalRedirectOnce( url ) {
```
Comment (seear) [inline client/jetpack-connect/authorize.js]: Nice.

**Comment by seear** [comment]
We could use `externalRedirect()` [here](https://github.com/Automattic/wp-calypso/pull/21104/files#diff-ff76e1701bebf84cf3e5a847ffd165d0R173) for consistency.

File: client/jetpack-connect/plans.jsx [hunk 2]
```diff
@@ -69,7 +70,7 @@ class Plans extends Component {
 
 	maybeRedirect() {
 		if ( this.props.isAutomatedTransfer ) {
-			this.props.goBackToWpAdmin( this.props.selectedSite.URL + JETPACK_ADMIN_PATH );
+			externalRedirect( this.props.selectedSite.URL + JETPACK_ADMIN_PATH );
```
Comment (seear) [inline client/jetpack-connect/plans.jsx]: I was testing this case and thinking about changing it to redirect to calypso plans. But looking back at the original [PR](https://github.com/Automattic/wp-calypso/pull/12695), it seems that this is for when AT sites have somehow become disconnected, and the remedy is to go through the jetpack connect procedure. So perhaps going to wpadmin does make sense in this context.

Suggest we add a comment such as `// When reconnecting broken AT sites`.

Does this make sense @artpi?

**Comment by sirreal** [comment]
Rebased to fix conflicts.

**Comment by sirreal** [comment]
> We could use `externalRedirect()` [here](https://github.com/Automattic/wp-calypso/pull/21104/files#diff-ff76e1701bebf84cf3e5a847ffd165d0R173) for consistency.

Slight difference that `replace()` won't add to the history stack (can't navigate back). I'm not sure if that has any relevance for mobile flows, I think they'll be closing the webview as soon as the navigation occurs either via `replace()` or assignment. I'm happy either way. See https://github.com/Automattic/wp-calypso/pull/20955#discussion_r158310041

**Comment by seear** [comment]
> I'm not sure if that has any relevance for mobile flows, I think they'll be closing the webview as soon as the navigation occurs either via replace() or assignment.

They also mentioned that the plan is to remove the back button from the webview.

**Comment by sirreal** [comment]
Thanks @seear!

I gave the xmlrpc another test and it continues to perform as expected. I'll merge this soon.

**Pull request merged**